File: package/umt_python/tests/unit/url/test_is_absolute_url.py
import unittest

from src.url import is_absolute_url


class TestIsAbsoluteUrl(unittest.TestCase):
    def test_http_urls(self):
        self.assertTrue(is_absolute_url("http://example.com"))
        self.assertTrue(is_absolute_url("https://example.com"))

    def test_other_schemes(self):
        self.assertTrue(is_absolute_url("ftp://files.example"))
        self.assertTrue(is_absolute_url("mailto:user@host"))
        self.assertTrue(is_absolute_url("tel:+1234567890"))
        self.assertTrue(is_absolute_url("ssh://host"))

    def test_custom_schemes(self):
        self.assertTrue(is_absolute_url("custom+scheme://path"))
        self.assertTrue(is_absolute_url("my-app://deep-link"))
        self.assertTrue(is_absolute_url("x.y://test"))

    def test_relative_paths(self):
        self.assertFalse(is_absolute_url("/path/to/page"))
        self.assertFalse(is_absolute_url("relative/path"))
        self.assertFalse(is_absolute_url("./relative"))
        self.assertFalse(is_absolute_url("../parent"))

    def test_protocol_relative_urls(self):
        self.assertFalse(is_absolute_url("//example.com"))

    def test_empty_string(self):
        self.assertFalse(is_absolute_url(""))

    def test_invalid_scheme_starts(self):
        self.assertFalse(is_absolute_url("123://invalid"))
        self.assertFalse(is_absolute_url("+bad://invalid"))


if __name__ == "__main__":
    unittest.main()

File: package/umt_rust/tests/url/test_is_absolute_url.rs
use umt_rust::url::umt_is_absolute_url;

#[test]
fn test_returns_true_for_http_urls() {
    assert!(umt_is_absolute_url("http://example.com"));
    assert!(umt_is_absolute_url("https://example.com"));
}

#[test]
fn test_returns_true_for_other_schemes() {
    assert!(umt_is_absolute_url("ftp://files.example"));
    assert!(umt_is_absolute_url("mailto:user@host"));
    assert!(umt_is_absolute_url("tel:+1234567890"));
    assert!(umt_is_absolute_url("ssh://host"));
}

#[test]
fn test_returns_true_for_custom_schemes() {
    assert!(umt_is_absolute_url("custom+scheme://path"));
    assert!(umt_is_absolute_url("my-app://deep-link"));
    assert!(umt_is_absolute_url("x.y://test"));
}

#[test]
fn test_returns_false_for_relative_paths() {
    assert!(!umt_is_absolute_url("/path/to/page"));
    assert!(!umt_is_absolute_url("relative/path"));
    assert!(!umt_is_absolute_url("./relative"));
    assert!(!umt_is_absolute_url("../parent"));
}

#[test]
fn test_returns_false_for_protocol_relative_urls() {
    assert!(!umt_is_absolute_url("//example.com"));
}

#[test]
fn test_returns_false_for_empty_string() {
    assert!(!umt_is_absolute_url(""));
}

#[test]
fn test_returns_false_for_invalid_scheme_starts() {
    assert!(!umt_is_absolute_url("123://invalid"));
    assert!(!umt_is_absolute_url("+bad://invalid"));
}
